### PR TITLE
test(spanner): integration tests for generated Spanner admin APIs

### PR DESCRIPTION
**Pull request by devbww**
Convert the g/c/spanner/admin integration tests to start using the
generated admin APIs.  This makes things only a little more verbose
as the handwrought APIs weren't that thick in the first place.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7302)

### CODE REVIEW

**Review by coryan — COMMENTED** [review]
With this change, the hand-crafted admin APIs do not have integration tests anymore.  They are used in the examples, but not tested beyond "it does not create an error".  That seems fine if we were planning to remove the hand-crafted admin APIs in a few weeks, but not if (as I think we should) we need to keep them around for a year after deprecation.

Can you walk me through the intended path here? Maybe it is written down somewhere?

- When are we planning to mark the hand-crafted admin APIs as "deprecated"?
- When are we planning to remove the hand-crafted admin APIs?
- How would we test the hand-crafted admin APIs between those two events?
- Would it save testing effort if we implemented the hand-crafted APIs using the automatically generated classes?

**Comment by devbww** [comment]
Sorry it wasn't clearer.

> With this change, the hand-crafted admin APIs do not have integration tests anymore.

They do.  Those tests are in g/c/spanner/integration_tests, whereas these ones are in g/c/spanner/**admin**/integration_tests.

The two APIs, and their tests, can peacefully co-exist.  We just won't be adding new features to the hand-crafted APIs (should they arise).

> When are we planning to mark the hand-crafted admin APIs as "deprecated"?

Sometime soon ... when we're happy that the generated ones are OK.  Certainly before the end of the month, I hope.

> When are we planning to remove the hand-crafted admin APIs?

One year after deprecation, I'd suggest.

> How would we test the hand-crafted admin APIs between those two events?

With their existing unit and integration tests.  That is, just the same way we do today, but excluding the samples, which will simply migrate.

> Would it save testing effort if we implemented the hand-crafted APIs using the automatically generated classes?

I don't think so.

**Review by coryan — APPROVED** [review]
I am not sure how I missed that these were new tests.  I confused myself because the code was mostly deltas as opposed to additions.

File: google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc [hunk 1]
```diff
@@ -203,8 +204,8 @@ TEST_F(InstanceAdminClientTest, InstanceConfig) {
 
   auto instance_config_names = [&project_id, this]() mutable {
     std::vector<std::string> names;
-    for (auto const& instance_config :
-         client_.ListInstanceConfigs(project_id)) {
+    auto const parent = "projects/" + project_id;
+    for (auto const& instance_config : client_.ListInstanceConfigs(parent)) {
```
[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc:208]: The customer would have to switch `client_` from `google::cloud::spanner::InstanceAdminClient` to `google::cloud::spanner_admin::InstanceAdminClient`, at which point they would also need to adapt to the different API.

If they didn't do that in this particular case, I believe they would get a clear status about a malformed project name.

**Comment by devbww** [comment]
> I am not sure how I missed that these were new tests. I confused myself because the code was mostly deltas as opposed to additions.

Yeah ... I understand.  Confusing because I did it that way to try to make the actual edits non-confusing!

**Pull request merged**